Email.Service/Email.Service.BLL/Handlers/MessageHandlerFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Email.Service.BLL.Interfaces;
+using Microsoft.Extensions.DependencyInjection;
+using SharingMessages;
+
+namespace Email.Service.BLL.Handlers;
+
+public class MessageHandlerFactory(IServiceProvider serviceProvider) : IMessageHandlerFactory
+{
+    public IMessageHandler CreateHandler(MessageType messageType)
+    {
+        return messageType switch
```

Comment:
Mb it would be better to create a strategies with base sendMessage Method and build in each strategy each type of message and call base SendMessage method? We can discuss it in office if you're interesting.